src/components/Transfer/Source.tsx

```diff
@@ -153,37 +148,25 @@ function Source() {
     dispatch(incrementStep());
   }, [dispatch]);
 
-  /* Only allow sending from ETH <-> BSC Pandle Token */
-  const [isPandle, setIsPandle] = useState(false);
   const selectedTokenAddress = useSelector(
     (state: RootState) => state.transfer.sourceParsedTokenAccount?.mintKey
   );
-  useEffect(() => {
-    const EthereumPandleAddress =
-      "0x808507121b80c02388fad14726482e061b8da827".toUpperCase();
-    const BscPandleAddres =
-      "0xb3Ed0A426155B79B898849803E3B36552f7ED507".toUpperCase();
-    const isFromEthereum = (
-      sourceChain: number,
-      selectedTokenAddress: string | undefined
-    ) =>
-      sourceChain === CHAIN_ID_ETH &&
-      selectedTokenAddress === EthereumPandleAddress;
-    const isFromBsc = (
-      sourceChain: number,
-      selectedTokenAddress: string | undefined
-    ) =>
-      sourceChain === CHAIN_ID_BSC && selectedTokenAddress === BscPandleAddres;
-    if (isFromEthereum(sourceChain, selectedTokenAddress?.toUpperCase())) {
-      setIsPandle(true);
-      handleTargetChange({ target: { value: CHAIN_ID_BSC } });
-    } else if (isFromBsc(sourceChain, selectedTokenAddress?.toUpperCase())) {
-      setIsPandle(true);
-      handleTargetChange({ target: { value: CHAIN_ID_ETH } });
-    } else {
-      setIsPandle(false);
-    }
-  }, [sourceChain, selectedTokenAddress, handleTargetChange]);
+  const { isTransferDisabled, warnings, ids } = useTransferControl(
+    TransferRules,
+    sourceChain,
+    targetChain,
+    selectedTokenAddress
+  );
+  /* Only allow sending from ETH <-> BSC Pandle Token */
+  const isPandle = (id: string) => id === "pandle";
+  const isRoundTripTransfer = useRoundTripTransfer(
```

Comment:
I'm failing to grasp the purpose of `useRoundTripTransfer`. Would it be to force the chain id to be BSC when picking ETH and selecting the Pandle token?

Reply:
Yes, the intention of it is to set up a tight logic to limit round trips between two chains. This is configured particularly for Pandle, but it could be used in other scenarios as well. This limit the target chain to be one specific based on the source selected.

Reply:
anything else here @M-Picco, is it enough clear to you?